hoot-js/src/main/cpp/hoot/js/conflate/matching/ScriptMatchCreator.cpp

```diff
@@ -612,18 +697,37 @@ void ScriptMatchCreator::createMatches(
   LOG_INFO(
     "Looking for matches with: " << className() << ";" << scriptFileInfo.fileName() << "...");
   LOG_VARD(*threshold);
-  // TODO: Can we filter the element types we iterate over here this based on the script being run?
-  map->visitRo(v);
-  // total match count inaccurate here
-//  LOG_INFO(
-//    "Found " << StringUtils::formatLargeNumber(v.getNumMatchCandidatesFound()) << " " <<
-//    CreatorDescription::baseFeatureTypeToString(scriptInfo.baseFeatureType) <<
-//    " match candidates and " << StringUtils::formatLargeNumber(matches.size()) <<
-//    " total matches in: " << StringUtils::millisecondsToDhms(timer.elapsed()) << ".");
+  LOG_VARD(GeometryTypeCriterion::typeToString(scriptInfo.geometryType));
+  switch (scriptInfo.geometryType)
```

Comment:
Optimized the match candidate consideration to only check what's necessary based on the conflate script's geometry type